frontend/test/pytest/test_callback.py

```diff
@@ -233,7 +233,13 @@ def identity(arg) -> int:
     def cir(x):
         return identity(x)
 
-    with pytest.raises(RuntimeError, match="TypeError: Callback identity expected type"):
+    # NOTE: Currently, this will raise a TypeError exception on Linux and the same exception
+    # wrapped as a RuntimeError on macOS. This appears to be related to using nanobind for the
+    # Python/C++ bindings. To avoid separate test cases for Linux and macOS, we accept either
+    # exception type here and match on the string below, which should be contained in the messages
+    # of both.
+    # TODO: Why does this happen?
```

Comment:
Great question 😅

Reply:
I think this just might be one of the quirks with nanobind, I don't see a good way around it at the moment :confused: